test/cppsuite/test_harness/test.h

```diff
@@ -0,0 +1,160 @@
+/*-
+ * Public Domain 2014-present MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef TEST_H
+#define TEST_H
+
+#include <vector>
+#include <mutex>
+
+extern "C" {
+#include "wiredtiger.h"
+}
+
+#include "api_const.h"
+#include "component.h"
+#include "configuration_settings.h"
+#include "connection.h"
+#include "runtime_monitor.h"
+#include "timestamp_manager.h"
+#include "thread_manager.h"
+#include "workload_generator.h"
+
+namespace test_harness {
+/*
+ * The base class for a test, the standard usage pattern is to just call run().
+ */
+class test {
+    public:
+    test(const std::string &config, bool enable_tracking)
+    {
+        _configuration = new configuration(name, config);
+        _workload_generator = new workload_generator(_configuration, enable_tracking);
+        _runtime_monitor = new runtime_monitor();
+        _timestamp_manager = new timestamp_manager();
+        _thread_manager = new thread_manager();
+        /*
+         * Ordering is not important here, any dependencies between components should be resolved
+         * internally by the components.
```

Comment:
What do you mean by internally? How that is supposed to work?

Reply:
A component shouldn't be built that it has a dependency on any other component, however if it does it needs to assert that that dependency is met prior to doing what it was planning on doing. I'm hoping we can avoid controlling ordering.